scripts/Engineering/common/calibration_info.py

```diff
@@ -215,25 +252,43 @@ def get_subplot_title(self, ispec):
         else:
             return f"{self.get_group_description()} spec: {ispec}"  # texture
 
-    def get_group_ws(self):
+    def get_group_ws(self) -> GroupingWorkspace:
         """
         Returns grouping workspace for ROI (creates if not present)
         :return: group workspace
         """
         if not self.group_ws or not ADS.doesExist(self.group_ws.name()):
+            self.update_group_ws_from_group()
+        return self.group_ws
+
+    def update_group_ws_from_group(self) -> None:
+        if self.group:
             if self.group.banks:
                 self.create_bank_grouping_workspace()
             elif self.group == GROUP.CROPPED:
                 self.create_grouping_workspace_from_spectra_list()
             elif self.group == GROUP.CUSTOM:
-                self.create_grouping_workspace_from_calfile()
-        return self.group_ws
+                self.create_grouping_workspace_from_file()
+
+    def create_grouping_workspace_from_file(self) -> None:
+        """
+        Create grouping workspace from a custom file (.cal or .xml)
+        """
+        if not self.grouping_filepath:
+            raise ValueError("Grouping file path is not set.")
+        ext = path.splitext(self.grouping_filepath)[-1].lower()
+        if ext == ".cal":
+            self.create_grouping_workspace_from_calfile()
```

Comment:
Do we need a separate function for this - not sure why it was like that in the first place (probably me!)
```suggestion
            grp_ws, _, _ = CreateGroupingWorkspace(InstrumentName=self.instrument, OldCalFilename=self.grouping_filepath, OutputWorkspace=self.get_group_ws_name()
```